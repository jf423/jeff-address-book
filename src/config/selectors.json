{
    "INDEX_PAGE": {
        "INDEX_CONTAINER": {
            "ID": "INDEX_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='INDEX_CONTAINER']",
            "EX": "[data-selector-id='INDEX_CONTAINER']",
            "PARAM": ""
        },
        "SEARCH_CONTAINER": {
            "ID": "SEARCH_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_CONTAINER']",
            "EX": "[data-selector-id='SEARCH_CONTAINER']",
            "PARAM": ""
        },
        "SEARCH_INPUT": {
            "ID": "SEARCH_INPUT",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_INPUT']",
            "EX": "[data-selector-id='SEARCH_INPUT']",
            "PARAM": ""
        },
        "SEARCH_CLEAR": {
            "ID": "SEARCH_CLEAR",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_CLEAR']",
            "EX": "[data-selector-id='SEARCH_CLEAR']",
            "PARAM": ""
        },
        "SEARCH_BUTTON": {
            "ID": "SEARCH_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='SEARCH_BUTTON']",
            "EX": "[data-selector-id='SEARCH_BUTTON']",
            "PARAM": ""
        },
        "USER_CONTAINER": {
            "ID": "USER_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CONTAINER']",
            "EX": "[data-selector-id='USER_CONTAINER']",
            "PARAM": ""
        },
        "USER_CONTENT": {
            "ID": "USER_CONTENT",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CONTENT']",
            "EX": "[data-selector-id='USER_CONTENT']",
            "PARAM": ""
        },
        "USER_CATALOG_END": {
            "ID": "USER_CATALOG_END",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CATALOG_END']",
            "EX": "[data-selector-id='USER_CATALOG_END']",
            "PARAM": ""
        },
        "USER_CELL_CONTAINER": {
            "ID": "USER_CELL_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1']",
            "PARAM": ""
        },
        "USER_CELL_CONTENT": {
            "ID": "USER_CELL_CONTENT",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_CONTENT']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_CONTENT']",
            "PARAM": ""
        },
        "USER_CELL_AVATAR": {
            "ID": "USER_CELL_AVATAR",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_AVATAR']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_AVATAR']",
            "PARAM": ""
        },
        "USER_CELL_NAME": {
            "ID": "USER_CELL_NAME",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_NAME']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_NAME']",
            "PARAM": ""
        },
        "USER_CELL_NATIONALITY": {
            "ID": "USER_CELL_NATIONALITY",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_NATIONALITY']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_NATIONALITY']",
            "PARAM": ""
        },
        "USER_CELL_USERNAME": {
            "ID": "USER_CELL_USERNAME",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_USERNAME']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_USERNAME']",
            "PARAM": ""
        },
        "USER_CELL_EMAIL": {
            "ID": "USER_CELL_EMAIL",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_CELL_CONTAINER_{INDEX}'] [data-selector-id='USER_CELL_EMAIL']",
            "EX": "[data-selector-id='USER_CELL_CONTAINER_1'] [data-selector-id='USER_CELL_EMAIL']",
            "PARAM": ""
        },
        "USER_MODAL_CONTAINER": {
            "ID": "USER_MODAL_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CONTAINER']",
            "EX": "[data-selector-id='USER_MODAL_CONTAINER']",
            "PARAM": ""
        },
        "USER_MODAL_CLOSE_BUTTON": {
            "ID": "USER_MODAL_CLOSE_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CLOSE_BUTTON']",
            "EX": "[data-selector-id='USER_MODAL_CLOSE_BUTTON']",
            "PARAM": ""
        },
        "USER_MODAL_AVATAR": {
            "ID": "USER_MODAL_AVATAR",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_AVATAR']",
            "EX": "[data-selector-id='USER_MODAL_AVATAR']",
            "PARAM": ""
        },
        "USER_MODAL_NAME": {
            "ID": "USER_MODAL_NAME",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_NAME']",
            "EX": "[data-selector-id='USER_MODAL_NAME']",
            "PARAM": ""
        },
        "USER_MODAL_STREET": {
            "ID": "USER_MODAL_STREET",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_STREET']",
            "EX": "[data-selector-id='USER_MODAL_STREET']",
            "PARAM": ""
        },
        "USER_MODAL_CITY": {
            "ID": "USER_MODAL_CITY",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CITY']",
            "EX": "[data-selector-id='USER_MODAL_CITY']",
            "PARAM": ""
        },
        "USER_MODAL_STATE": {
            "ID": "USER_MODAL_STATE",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_STATE']",
            "EX": "[data-selector-id='USER_MODAL_STATE']",
            "PARAM": ""
        },
        "USER_MODAL_POSTCODE": {
            "ID": "USER_MODAL_POSTCODE",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_POSTCODE']",
            "EX": "[data-selector-id='USER_MODAL_POSTCODE']",
            "PARAM": ""
        },
        "USER_MODAL_PHONE": {
            "ID": "USER_MODAL_PHONE",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_PHONE']",
            "EX": "[data-selector-id='USER_MODAL_PHONE']",
            "PARAM": ""
        },
        "USER_MODAL_CELL": {
            "ID": "USER_MODAL_CELL",
            "TYPE": "css",
            "VAL": "[data-selector-id='USER_MODAL_CELL']",
            "EX": "[data-selector-id='USER_MODAL_CELL']",
            "PARAM": ""
        }
    },
    "SETTINGS_PAGE": {
        "SETTINGS_CONTAINER": {
            "ID": "SETTINGS_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='SETTINGS_CONTAINER']",
            "EX": "[data-selector-id='SETTINGS_CONTAINER']",
            "PARAM": ""
        },
        "NATIONALITIES_CONTAINER": {
            "ID": "NATIONALITIES_CONTAINER",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CONTAINER']",
            "EX": "[data-selector-id='NATIONALITIES_CONTAINER']",
            "PARAM": ""
        },
        "NATIONALITIES_BACK_BUTTON": {
            "ID": "NATIONALITIES_BACK_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_BACK_BUTTON']",
            "EX": "[data-selector-id='NATIONALITIES_BACK_BUTTON']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_CH": {
            "ID": "NATIONALITIES_CHECKBOX_CH",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_CH']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_CH']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_ES": {
            "ID": "NATIONALITIES_CHECKBOX_ES",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_ES']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_ES']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_FR": {
            "ID": "NATIONALITIES_CHECKBOX_FR",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_FR']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_FR']",
            "PARAM": ""
        },
        "NATIONALITIES_CHECKBOX_GB": {
            "ID": "NATIONALITIES_CHECKBOX_GB",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_CHECKBOX_GB']",
            "EX": "[data-selector-id='NATIONALITIES_CHECKBOX_GB']",
            "PARAM": ""
        },
        "NATIONALITIES_SAVE_BUTTON": {
            "ID": "NATIONALITIES_SAVE_BUTTON",
            "TYPE": "css",
            "VAL": "[data-selector-id='NATIONALITIES_SAVE_BUTTON']",
            "EX": "[data-selector-id='NATIONALITIES_SAVE_BUTTON']",
            "PARAM": ""
        }
    }
}