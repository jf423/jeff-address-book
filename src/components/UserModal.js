import PropTypes from 'prop-types';
import styled from 'styled-components';

import { IconButton, Modal } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import { flexRowCenterCenter, flexColCenterCenter } from '@/style/flex.css.js';
import { eclipse, shadow } from '@/style/common.css.js';

import { INDEX_PAGE } from '@/config/selectors.json';

const {
    USER_MODAL_CONTAINER,
    USER_MODAL_CLOSE_BUTTON,
    USER_MODAL_AVATAR,
    USER_MODAL_NAME,
    USER_MODAL_STREET,
    USER_MODAL_CITY,
    USER_MODAL_STATE,
    USER_MODAL_POSTCODE,
    USER_MODAL_PHONE,
    USER_MODAL_CELL
} = INDEX_PAGE;

const UserModal = ({ open, selectedUser, setSelectedIndex }) => {
    const onCloseModal = () => {
        setSelectedIndex(null);
    };

    return (
        <Container
            data-selector-id={USER_MODAL_CONTAINER.ID}
            disableEnforceFocus
            disableAutoFocus
            open={open}
            onClose={onCloseModal}
            aria-labelledby={`${selectedUser?.name?.first} ${selectedUser?.name?.last}`}
        >
            <ModalContent>
                <CloseModalButton data-selector-id={USER_MODAL_CLOSE_BUTTON.ID} onClick={onCloseModal} aria-label="close modal">
                    <CloseIcon />
                </CloseModalButton>
                <Avatar data-selector-id={USER_MODAL_AVATAR.ID} src={selectedUser?.picture?.large} />
                <Name data-selector-id={USER_MODAL_NAME.ID}>
                    {`${selectedUser?.name?.first} ${selectedUser?.name?.last}`}
                </Name>
                <Info>
                    <Text data-selector-id={USER_MODAL_STREET.ID}>
                        {`Street: ${selectedUser?.location?.street?.number} ${selectedUser?.location?.street?.name}`}
                    </Text>
                    <Text data-selector-id={USER_MODAL_CITY.ID}>
                        {`City: ${selectedUser?.location?.city}`}
                    </Text>
                    <Text data-selector-id={USER_MODAL_STATE.ID}>
                        {`State: ${selectedUser?.location?.state}`}
                    </Text>
                    <Text data-selector-id={USER_MODAL_POSTCODE.ID}>
                        {`Postcode: ${selectedUser?.location?.postcode}`}
                    </Text>
                    <Text data-selector-id={USER_MODAL_PHONE.ID}>
                        {`Phone: ${selectedUser?.phone}`}
                    </Text>
                    <Text data-selector-id={USER_MODAL_CELL.ID}>
                        {`Cell: ${selectedUser?.cell}`}
                    </Text>
                </Info>
            </ModalContent>
        </Container>
    );
};

export default UserModal;

UserModal.defaultProps = {
    open: false,
    selectedUser: null
};

UserModal.propTypes = {
    open: PropTypes.bool,
    selectedUser: PropTypes.object,
    setSelectedIndex: PropTypes.func.isRequired
};

const Avatar = styled.img`
    border: 5px solid #fff;
    background: #ddd;
    border-radius: 100%;
    width: 100px;
    height: 100px;
`;

const Name = styled.h2`
    text-align: center;
    max-width: 100%;
    ${eclipse};
`;

const Info = styled.div`
    width: 100%;
    min-height: 80px;
    ${flexColCenterCenter};
`;

const Text = styled.span`
    max-width: 100%;
    padding: 5px;
    color: #6c757d;
    ${eclipse};
`;

const CloseModalButton = styled(IconButton)``;

const Container = styled(Modal)`
    ${flexRowCenterCenter};
`;

const ModalContent = styled.div`
    position: relative;
    width: auto;
    height: auto;
    padding: 20px;
    background: #fff;
    ${shadow};
    ${flexColCenterCenter};

    ${Avatar} {
        width: 250px;
        height: 250px;
    }
    ${CloseModalButton} {
        position: absolute;
        top: 10px;
        right: 10px;
    }
`;
