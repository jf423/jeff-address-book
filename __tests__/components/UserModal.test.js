import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { toBeInTheDocument, toHaveTextContent, toHaveValue } from '@testing-library/jest-dom/matchers';

import { INDEX_PAGE } from '@/config/selectors.json';

import UserModal from '@/components/UserModal.js';

import DATA from '../mockUsersData.json';

expect.extend({ toBeInTheDocument, toHaveTextContent, toHaveValue });

const mockData = {
    open: true,
    selectedUser: DATA.results[0],
    setSelectedIndex: jest.fn()
};
const {
    USER_MODAL_CONTAINER,
    USER_MODAL_CLOSE_BUTTON,
    USER_MODAL_AVATAR,
    USER_MODAL_NAME,
    USER_MODAL_STREET,
    USER_MODAL_CITY,
    USER_MODAL_STATE,
    USER_MODAL_POSTCODE,
    USER_MODAL_PHONE,
    USER_MODAL_CELL
} = INDEX_PAGE;

describe('UserModal component test', () => {
    let userModal;
    const renderUserModal = props => (
        <UserModal {...props} />
    );

    beforeEach(() => {
        userModal = render(renderUserModal(mockData));
    });

    afterEach(() => {
        userModal.unmount();
    });

    test('should display as expected', () => {
        const { selectedUser } = mockData;
        const userModalContainer = userModal.getByTestId(USER_MODAL_CONTAINER.ID);
        const userModalCloseButton = userModal.getByTestId(USER_MODAL_CLOSE_BUTTON.ID);
        const userModalAvatar = userModal.getByTestId(USER_MODAL_AVATAR.ID);
        const userModalName = userModal.getByTestId(USER_MODAL_NAME.ID);
        const userModalStreet = userModal.getByTestId(USER_MODAL_STREET.ID);
        const userModalCity = userModal.getByTestId(USER_MODAL_CITY.ID);
        const userModalState = userModal.getByTestId(USER_MODAL_STATE.ID);
        const userModalPostcode = userModal.getByTestId(USER_MODAL_POSTCODE.ID);
        const userModalPhone = userModal.getByTestId(USER_MODAL_PHONE.ID);
        const userModalCell = userModal.getByTestId(USER_MODAL_CELL.ID);

        expect(userModalContainer).toBeInTheDocument();
        expect(userModalCloseButton).toBeInTheDocument();
        expect(userModalAvatar.src).toBe(selectedUser?.picture?.large);
        expect(userModalName).toHaveTextContent(`${selectedUser?.name?.first} ${selectedUser?.name?.last}`);
        expect(userModalStreet).toHaveTextContent(`${selectedUser?.location?.street?.number} ${selectedUser?.location?.street?.name}`);
        expect(userModalCity).toHaveTextContent(selectedUser?.location?.city);
        expect(userModalState).toHaveTextContent(selectedUser?.location?.state);
        expect(userModalPostcode).toHaveTextContent(selectedUser?.location?.postcode);
        expect(userModalPhone).toHaveTextContent(selectedUser?.phone);
        expect(userModalCell).toHaveTextContent(selectedUser?.cell);
        expect(userModal).toMatchSnapshot();
    });

    test('should be trigger correctly when the user clicks the cell', () => {
        const userModalCloseButton = userModal.getByTestId(USER_MODAL_CLOSE_BUTTON.ID);

        fireEvent.click(userModalCloseButton);
        expect(mockData.setSelectedIndex).toHaveBeenCalledWith(null);
    });
});
