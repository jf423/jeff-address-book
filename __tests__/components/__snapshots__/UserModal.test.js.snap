// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModal component test should display as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      aria-labelledby="Mario Sanchez"
      class="UserModal__Container-nbddo3-5 khyklO"
      data-selector-id="USER_MODAL_CONTAINER"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="UserModal__ModalContent-nbddo3-6 geQXQP"
        tabindex="-1"
      >
        <button
          aria-label="close modal"
          class="MuiButtonBase-root MuiIconButton-root UserModal__CloseModalButton-nbddo3-4"
          data-selector-id="USER_MODAL_CLOSE_BUTTON"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <img
          class="UserModal__Avatar-nbddo3-0 kvxjCD"
          data-selector-id="USER_MODAL_AVATAR"
          src="https://randomuser.me/api/portraits/men/29.jpg"
        />
        <h2
          class="UserModal__Name-nbddo3-1 jajtsx"
          data-selector-id="USER_MODAL_NAME"
        >
          Mario Sanchez
        </h2>
        <div
          class="UserModal__Info-nbddo3-2 eZbDxY"
        >
          <span
            class="UserModal__Text-nbddo3-3 fkZDZk"
            data-selector-id="USER_MODAL_STREET"
          >
            Street: 9153 Paseo de Zorrilla
          </span>
          <span
            class="UserModal__Text-nbddo3-3 fkZDZk"
            data-selector-id="USER_MODAL_CITY"
          >
            City: Zaragoza
          </span>
          <span
            class="UserModal__Text-nbddo3-3 fkZDZk"
            data-selector-id="USER_MODAL_STATE"
          >
            State: Pa√≠s Vasco
          </span>
          <span
            class="UserModal__Text-nbddo3-3 fkZDZk"
            data-selector-id="USER_MODAL_POSTCODE"
          >
            Postcode: 38460
          </span>
          <span
            class="UserModal__Text-nbddo3-3 fkZDZk"
            data-selector-id="USER_MODAL_PHONE"
          >
            Phone: 966-530-154
          </span>
          <span
            class="UserModal__Text-nbddo3-3 fkZDZk"
            data-selector-id="USER_MODAL_CELL"
          >
            Cell: 618-277-571
          </span>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
